File: libs/interfaces/src/lib/loan.interface.ts
import { ClientInterface } from './client.interface';
import { CurrencyInterface } from './currency.interface';

export interface LoanInterface {
  id: number;
  purchaseDate: Date;
  currency: CurrencyInterface;
  purchaseValue: number;
  interestRate: number;
  dueDate: Date;
  client:ClientInterface;
  monthsCount: number;
  finalAmount: number;
  conversionRate: number;
}

File: libs/java-interfaces/src/main/java/com/loanmanager/interfaces/LoanInterface.java
package com.loanmanager.interfaces;

import java.time.LocalDate;

public interface LoanInterface {
    Long getId();
    void setId(Long id);
    
    LocalDate getPurchaseDate();
    void setPurchaseDate(LocalDate purchaseDate);
    
    String getCurrencyType();
    void setCurrencyType(String currencyType);
    
    Double getPurchaseValue();
    void setPurchaseValue(Double purchaseValue);
    
    Double getConversionRate();
    void setConversionRate(Double conversionRate);
    
    Double getFinalAmount();
    void setFinalAmount(Double finalAmount);
    
    Double getInterestRate();
    void setInterestRate(Double interestRate);
    
    LocalDate getDueDate();
    void setDueDate(LocalDate dueDate);
    
    Integer getMonthsCount();
    void setMonthsCount(Integer monthsCount);
    
    ClientInterface getClient();
    void setClient(ClientInterface client);
}